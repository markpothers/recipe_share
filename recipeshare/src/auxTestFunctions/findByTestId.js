export const findByTestID = (component, type, testId) => {
	return component.find(type).filterWhere(c => c.props().testID === testId).first().props()
}