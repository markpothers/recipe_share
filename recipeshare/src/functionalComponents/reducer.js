const reducer = (currentState, action) => {
	switch (action.type) {
		case 'STORE_RECIPE_LISTS':
			// console.log("saving recipes list")
			return { ...currentState, recipes: { ...currentState.recipes, [action.recipeType]: action.recipeList } }
		// case 'CLEAR_LISTED_RECIPES':
		//     // console.log("clearing listed recipes")
		//     return {...currentState, recipes: {...currentState.recipes, [action.recipeType]: []}}
		case 'STORE_RECIPE_DETAILS':
			// console.log(action.recipe_details)
			// console.log(action.recipe_details)
			return { ...currentState, recipe_details: action.recipe_details }
		case 'APPEND_TO_RECIPE_LISTS':
			// console.log(currentState.recipes[action.recipeType])
			return { ...currentState, recipes: { ...currentState.recipes, [action.recipeType]: [...currentState.recipes[action.recipeType], ...action.recipeList] } }
		case 'APPEND_TO_RECIPES_DETAILS':
			// console.log("appending recipes details")
			// console.log(currentState.recipes_details[action.recipeType].comments)
			// console.log(action.recipesDetailsList)
			return {
				...currentState, recipes_details: {
					...currentState.recipes_details, [action.recipeType]: {
						comments: [...currentState.recipes_details[action.recipeType].comments, ...action.recipesDetailsList.comments],
						ingredient_uses: [...currentState.recipes_details[action.recipeType].ingredient_uses, ...action.recipesDetailsList.ingredient_uses],
						ingredients: [...currentState.recipes_details[action.recipeType].ingredients, ...action.recipesDetailsList.ingredients],
						make_pics: [...currentState.recipes_details[action.recipeType].make_pics, ...action.recipesDetailsList.make_pics],
						recipe_images: [...currentState.recipes_details[action.recipeType].recipe_images, ...action.recipesDetailsList.recipe_images],
						recipe_likes: [...currentState.recipes_details[action.recipeType].recipe_likes, ...action.recipesDetailsList.recipe_likes],
						recipe_makes: [...currentState.recipes_details[action.recipeType].recipe_makes, ...action.recipesDetailsList.recipe_makes],
						recipes: [...currentState.recipes_details[action.recipeType].recipes, ...action.recipesDetailsList.recipes],
					}
				}
			}
		case 'UPDATE_NEW_USER_DETAILS':
			return { ...currentState, newUserDetails: { ...currentState.newUserDetails, [action.parameter]: action.content } }
		case 'CLEAR_NEW_USER_DETAILS':
			// console.log("clearing new user details")
			return {
				...currentState, newUserDetails: {
					first_name: "",
					last_name: "",
					username: "",
					e_mail: "",
					password: "",
					password_confirmation: "",
					profile_text: "",
					country: "United States",
					image_url: ""
				}
			}
		case 'UPDATE_LOGIN_USER_DETAILS':
			return { ...currentState, loginUserDetails: { ...currentState.loginUserDetails, [action.parameter]: action.content } }
		case 'CLEAR_LOGIN_USER_DETAILS':
			// console.log("clearing login details")
			return {
				...currentState, loginUserDetails: {
					e_mail: "",
					password: "",
				}
			}
		case 'UPDATE_NEW_RECIPE_DETAILS':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, [action.parameter]: action.content } }
		case 'UPDATE_RECIPE_INGREDIENTS':
			return {
				...currentState, newRecipeDetails: {
					...currentState.newRecipeDetails, ingredients: {
						...currentState.newRecipeDetails.ingredients, [action.ingredientIndex]: {
							name: action.ingredientName,
							quantity: action.ingredientQuantity,
							unit: action.ingredientUnit
						}
					}
				}
			}
		case 'STORE_ALL_INGREDIENTS':
			// console.log(action.ingredients)
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, ingredients: action.ingredients } }
		case 'CLEAR_NEW_RECIPE_DETAILS':
			// console.log("clearing new recipe details")
			return {
				...currentState, newRecipeDetails: {
					name: "",
					instructions: [
						'Pre heat oven to 450F...',
						'Dice the chicken...',
						'Add the onion...',
						'',
					],
					instructionImages: [],
					ingredients: {
						ingredient1: {
							name: "",
							quantity: "",
							unit: "Oz"
						}
					},
					difficulty: "0",
					time: "00:15",
					imageBase64: "",
					filter_settings: {
						"Breakfast": false,
						"Lunch": false,
						"Dinner": false,
						"Chicken": false,
						"Red meat": false,
						"Seafood": false,
						"Vegetarian": false,
						"Salad": false,
						"Vegan": false,
						"Soup": false,
						"Dessert": false,
						"Side": false,
						"Whole 30": false,
						"Paleo": false,
						"Freezer meal": false,
						"Keto": false,
						"Weeknight": false,
						"Weekend": false,
						"Gluten free": false,
						"Bread": false,
						"Dairy free": false,
						"White meat": false,
					},
					cuisine: "Any",
					serves: "Any",
					acknowledgement: ""
				}
			}
		// case 'LIKE_RECIPE':
		//         currentState.recipes[action.listType].find( recipe => recipe.id === recipeID)
		//     return {...currentState, recipes: {...currentState.recipes, [action.recipeType]: {...currentState.recipes[action.listType], }}}
		case 'ADD_RECIPE_LIKE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_likes: currentState.recipe_details.recipe_likes + 1, likeable: false } }
		case 'UPDATE_COMMENTS':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, comments: action.comments } }
		case 'REMOVE_RECIPE_LIKE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_likes: currentState.recipe_details.recipe_likes - 1, likeable: true } }
		case 'ADD_RECIPE_MAKE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_makes: currentState.recipe_details.recipe_makes + 1, makeable: false } }
		case 'ADD_RECIPE_SHARE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_shares: currentState.recipe_details.recipe_shares + 1, shareable: false } }
		case 'REMOVE_RECIPE_SHARE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_shares: currentState.recipe_details.recipe_shares - 1, shareable: true } }
		case 'ADD_MAKE_PIC':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, make_pics: [action.makePic, ...currentState.recipe_details.make_pics] } }
		case 'SAVE_REMAINING_MAKE_PICS':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, make_pics: action.makePics } }
		case 'REMOVE_RECIPE_LIKES':
			return { ...currentState, recipes_details: { ...currentState.recipes_details, [action.listType]: { ...currentState.recipes_details[action.listType], recipe_likes: action.recipe_likes } } }
		case 'STORE_NEW_FOLLOWERS':
			return { ...currentState, chefs_details: { ...currentState.chefs_details, [action.chefID]: { ...currentState.chefs_details[action.chefID], followers: action.followers, chef_followed: !currentState.chefs_details[action.chefID].chef_followed } } }
		// case 'UPDATE_CHEF_IN_LIST':
		//     return {...currentState, chefs: {...currentState.chefs, [action.chefType]: action.chefList}}
		case 'LOG_IN_CHEF':
			// console.log("logging in chef")
			return {
				...currentState, loggedInChef: {
					id: action.id,
					username: action.username,
				}
			}
		case 'UPDATE_LOGGED_IN_CHEF':
			// console.log("updating logged in chef")
			return {
				...currentState, loggedInChef: {
					id: action.id,
					username: action.username,
					auth_token: action.auth_token,
					image_url: action.image_url,
					is_admin: action.is_admin
				}
			}
		// case 'CHANGE_GLOBAL_RANKING':
		//         // console.log("switching ranking")
		//         const newValue = currentState.global_ranking == "liked" ? "made" : "liked"
		//         return {...currentState, global_ranking: newValue}
		case 'STORE_CHEF_LIST':
			// console.log("saving chefs list")
			return { ...currentState, chefs: { ...currentState.chefs, [action.chefType]: action.chefList } }
		case 'CLEAR_LISTED_CHEFS':
			// console.log("clearing listed chefs")
			return { ...currentState, chefs: { ...currentState.chefs, [action.chefType]: [] } }
		// case 'STORE_CHEFS_DETAILS':
		//     // console.log(action.chefsDetailsList)
		//     return {...currentState, chefs_details: {...currentState.chefs_details, [action.chefType]: action.chefsDetailsList}}
		case 'APPEND_TO_CHEF_LISTS':
			// console.log(currentState.chefs[action.chefType])
			return { ...currentState, chefs: { ...currentState.chefs, [action.chefType]: [...currentState.chefs[action.chefType], ...action.chefList] } }
		case 'STORE_CHEF_DETAILS':
			// console.log(action.chef_details)
			return { ...currentState, chefs_details: { ...currentState.chefs_details, [action.chefID]: action.chef_details } }
		case 'CLEAR_CHEF_DETAILS':
			// console.log(action.chef_details)
			return { ...currentState, chef_details: {} }
		case 'TOGGLE_RECIPES_LIST_FILTER_CATEGORY':
			// console.log(action.chef_details)
			return { ...currentState, filter_settings: { ...currentState.filter_settings, [action.category]: action.value } }
		case 'CLEAR_RECIPES_LIST_FILTERS':
			return { ...currentState, filter_settings: action.clearedFilters, cuisine: "Any", serves: "Any" }
		case 'SET_RECIPES_LIST_CUISINE':
			return { ...currentState, cuisine: action.cuisine }
		case 'SET_RECIPES_LIST_SERVES':
			return { ...currentState, serves: action.serves }
		case 'TOGGLE_NEW_RECIPE_FILTER_CATEGORY':
			// console.log(action.chef_details)
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, filter_settings: { ...currentState.newRecipeDetails.filter_settings, [action.category]: action.value } } }
		case 'CLEAR_NEW_RECIPE_FILTERS':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, filter_settings: action.clearedFilters, cuisine: "Any", serves: "Any" } }
		case 'SET_NEW_RECIPE_CUISINE':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, cuisine: action.cuisine } }
		case 'SET_NEW_RECIPE_SERVES':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, serves: action.serves } }
		case 'STAY_LOGGED_IN':
			return { ...currentState, stayLoggedIn: action.value }
		default:
			return currentState
	}
}

export default reducer
