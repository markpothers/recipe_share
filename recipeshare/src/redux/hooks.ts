import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux"
import type { AppDispatch } from "./store"
import type { RootStateType } from "./types"

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootStateType> = useSelector
