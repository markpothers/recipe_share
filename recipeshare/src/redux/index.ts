export { useAppDispatch, useAppSelector } from "./hooks";
export * from "./types";
export type { RootState } from "./store"
export {
	updateSingleRecipeList,
	updateAllRecipeLists,
	updateSingleChefList,
	updateAllChefLists,
	storeRecipeDetails,
	updateNewUserDetails,
	clearNewUserDetails,
	updateLoginUserDetails,
	clearLoginUserDetails,
	addRecipeLike,
	removeRecipeLike,
	addRecipeReShare,
	removeRecipeReShare,
	addRecipeMake,
	addMakePic,
	addMakePicChef,
	saveRemainingMakePics,
	updateComments,
	storeNewFollowers,
	updateLoggedInChef,
	storeChefDetails,
	clearChefDetails,
	stayLoggedIn
} from "./rootReducer"
export { default as rootReducer } from "./rootReducer"
