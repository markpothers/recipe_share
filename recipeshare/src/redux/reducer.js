const reducer = (currentState, action) => {
	switch (action.type) {
		case 'UPDATE_SINGLE_RECIPE_LIST':
			return { ...currentState, allRecipeLists: { ...currentState.allRecipeLists, [action.listKey]: action.recipeList}}
		case 'UPDATE_ALL_RECIPE_LISTS':
			return { ...currentState, allRecipeLists: action.allRecipeLists}
		case 'UPDATE_SINGLE_CHEF_LIST':
			return { ...currentState, allChefLists: { ...currentState.allChefLists, [action.listKey]: action.chefList}}
		case 'UPDATE_ALL_CHEF_LISTS':
			return { ...currentState, allChefLists: action.allChefLists}
		case 'STORE_RECIPE_DETAILS':
			return { ...currentState, recipe_details: action.recipe_details }
		case 'UPDATE_NEW_USER_DETAILS':
			return { ...currentState, newUserDetails: { ...currentState.newUserDetails, [action.parameter]: action.content } }
		case 'CLEAR_NEW_USER_DETAILS':
			// console.log("clearing new user details")
			return {
				...currentState, newUserDetails: {
					first_name: "",
					last_name: "",
					username: "",
					e_mail: "",
					password: "",
					password_confirmation: "",
					profile_text: "",
					country: "United States",
					image_url: ""
				}
			}
		case 'UPDATE_LOGIN_USER_DETAILS':
			return { ...currentState, loginUserDetails: { ...currentState.loginUserDetails, [action.parameter]: action.content } }
		case 'CLEAR_LOGIN_USER_DETAILS':
			// console.log("clearing login details")
			return {
				...currentState, loginUserDetails: {
					e_mail: "",
					password: "",
				}
			}
		case 'ADD_RECIPE_LIKE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_likes: currentState.recipe_details.recipe_likes + 1, likeable: false } }
		case 'UPDATE_COMMENTS':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, comments: action.comments } }
		case 'REMOVE_RECIPE_LIKE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_likes: currentState.recipe_details.recipe_likes - 1, likeable: true } }
		case 'ADD_RECIPE_MAKE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_makes: currentState.recipe_details.recipe_makes + 1, makeable: false } }
		case 'ADD_RECIPE_SHARE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_shares: currentState.recipe_details.recipe_shares + 1, shareable: false } }
		case 'REMOVE_RECIPE_SHARE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_shares: currentState.recipe_details.recipe_shares - 1, shareable: true } }
		case 'ADD_MAKE_PIC':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, make_pics: [action.makePic, ...currentState.recipe_details.make_pics] } }
		case 'ADD_MAKE_PIC_CHEF':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, make_pics_chefs: [action.makePicChef, ...currentState.recipe_details.make_pics_chefs] } }
		case 'SAVE_REMAINING_MAKE_PICS':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, make_pics: action.makePics } }
		case 'REMOVE_RECIPE_LIKES':
			return { ...currentState, recipes_details: { ...currentState.recipes_details, [action.listType]: { ...currentState.recipes_details[action.listType], recipe_likes: action.recipe_likes } } }
		case 'STORE_NEW_FOLLOWERS':
			return { ...currentState, chefs_details: { ...currentState.chefs_details, [action.chefID]: { ...currentState.chefs_details[action.chefID], followers: action.followers, chef_followed: !currentState.chefs_details[action.chefID].chef_followed } } }
		case 'LOG_IN_CHEF':
			// console.log("logging in chef")
			return {
				...currentState, loggedInChef: {
					id: action.id,
					username: action.username,
				}
			}
		case 'UPDATE_LOGGED_IN_CHEF':
			// console.log("updating logged in chef")
			return {
				...currentState, loggedInChef: {
					id: action.id,
					e_mail: action.e_mail,
					username: action.username,
					auth_token: action.auth_token,
					image_url: action.image_url,
					is_admin: action.is_admin
				}
			}
		case 'STORE_CHEF_LIST':
			return { ...currentState, chefs: { ...currentState.chefs, [action.chefType]: action.chefList } }
		case 'CLEAR_LISTED_CHEFS':
			return { ...currentState, chefs: { ...currentState.chefs, [action.chefType]: [] } }
		case 'APPEND_TO_CHEF_LISTS':
			// console.log(currentState.chefs[action.chefType])
			return { ...currentState, chefs: { ...currentState.chefs, [action.chefType]: [...currentState.chefs[action.chefType], ...action.chefList] } }
		case 'STORE_CHEF_DETAILS':
			// console.log(action.chef_details)
			return { ...currentState, chefs_details: { ...currentState.chefs_details, [action.chefID]: action.chef_details } }
		case 'CLEAR_CHEF_DETAILS':
			// console.log(action.chef_details)
			return { ...currentState, chef_details: {} }
		case 'TOGGLE_RECIPES_LIST_FILTER_CATEGORY':
			// console.log(action.chef_details)
			return { ...currentState, filter_settings: { ...currentState.filter_settings, [action.category]: action.value } }
		case 'CLEAR_RECIPES_LIST_FILTERS':
			return { ...currentState, filter_settings: action.clearedFilters, cuisine: "Any", serves: "Any" }
		case 'SET_RECIPES_LIST_CUISINE':
			// return { ...currentState, cuisine: action.cuisine }
			return { ...currentState, filterCuisines: { ...currentState.filterCuisines, [action.listChoice]: action.cuisine } }
		case 'SET_RECIPES_LIST_SERVES':
			return { ...currentState, serves: action.serves }
		case 'TOGGLE_NEW_RECIPE_FILTER_CATEGORY':
			// console.log(action.chef_details)
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, filter_settings: { ...currentState.newRecipeDetails.filter_settings, [action.category]: action.value } } }
		case 'CLEAR_NEW_RECIPE_FILTERS':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, filter_settings: action.clearedFilters, cuisine: "Any", serves: "Any" } }
		case 'SET_NEW_RECIPE_CUISINE':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, cuisine: action.cuisine } }
		case 'SET_NEW_RECIPE_SERVES':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, serves: action.serves } }
		case 'STAY_LOGGED_IN':
			return { ...currentState, stayLoggedIn: action.value }
		default:
			return currentState
	}
}

export default reducer
