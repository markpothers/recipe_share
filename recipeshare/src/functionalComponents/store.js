import { createStore, compose, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk'
import reducer from './reducer'

const initialState = {
  loggedInChef: {
    id: "",
    username: "",
    auth_token: "",
    imageURL: "",
    is_admin: false
  },
  recipes: {
    all: [],
    chef: [],
    chef_feed: [],
    chef_liked: [],
    chef_made: [],
    global_ranks: [],
    most_liked: [],
    most_made: []
  },
  recipes_details: {

  },
  newRecipeDetails: {
    name: "",
    instructions: ['test', 'test2'],
    ingredients: {
      ingredient1 :{
        name:"",
        quantity: "",
        unit: "Oz"
      }
    },
    difficulty: "0",
    time: "00:15",
    imageBase64: "",
    filter_settings: {
      "Breakfast": false,
      "Lunch": false,
      "Dinner": false,
      "Chicken": false,
      "Red meat": false,
      "Seafood": false,
      "Vegetarian": false,
      "Salad": false,
      "Vegan": false,
      "Soup": false,
      "Dessert": false,
      "Side": false,
      "Whole 30": false,
      "Paleo": false,
      "Freezer meal": false,
      "Keto": false,
      "Weeknight": false,
      "Weekend": false,
      "Gluten free": false,
      "Bread": false,
      "Dairy free": false,
      "White meat": false,
    },
    cuisine: "Any",
    serves: "Any",
    acknowledgement: ""
  },
  newUserDetails: {
    first_name: "",
    last_name: "",
    username: "",
    e_mail: "",
    password: "",
    password_confirmation: "",
    country: "United States",
    imageURL: "",
    profile_text: ""
  },
  loginUserDetails: {
    e_mail: "markpothers@hotmail.com",
    password: "123456"
  },
  chefs: {
    all_chefs: [],
    followed: [],
    most_liked_chefs: [],
    most_made_chefs: [],
    chef_followees: [],
    chef_followers: []
  },
  chefs_details: {},
  filter_settings: {
    "Breakfast": false,
    "Lunch": false,
    "Dinner": false,
    "Chicken": false,
    "Red meat": false,
    "Seafood": false,
    "Vegetarian": false,
    "Salad": false,
    "Vegan": false,
    "Soup": false,
    "Dessert": false,
    "Side": false,
    "Whole 30": false,
    "Paleo": false,
    "Freezer meal": false,
    "Keto": false,
    "Weeknight": false,
    "Weekend": false,
    "Gluten free": false,
    "Bread": false,
    "Dairy free": false,
    "White meat": false,
  },
  cuisine: "Any",
  serves: "Any",
  stayLoggedIn: false,
}

  const middleware = compose(
    applyMiddleware(ReduxThunk)
  )

  export const store = createStore(reducer, initialState, middleware)
