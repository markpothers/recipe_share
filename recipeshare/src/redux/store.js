import { createStore, compose, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk'
import reducer from './reducer'

// import { filters } from '../dataComponents/filters'

//exported ONLY for unit testing
export const initialState = {
	loggedInChef: {
		id: "",
		username: "",
		auth_token: "",
		image_url: "",
		is_admin: false
	},
	allRecipeLists: {},
	allChefLists: {},
	recipes_details: {

	},
	newRecipeDetails: {
		name: "",
		instructions: [
			'',
			'',
			'',
			'',
		],
		instructionImages: [],
		ingredients: {
			ingredient1: {
				name: "",
				quantity: "",
				unit: "Oz"
			}
		},
		difficulty: "0",
		time: "00:15",
		imageBase64: "",
		filter_settings: {
			"Breakfast": false,
			"Lunch": false,
			"Dinner": false,
			"Chicken": false,
			"Red meat": false,
			"Seafood": false,
			"Vegetarian": false,
			"Salad": false,
			"Vegan": false,
			"Soup": false,
			"Dessert": false,
			"Side": false,
			"Whole 30": false,
			"Paleo": false,
			"Freezer meal": false,
			"Keto": false,
			"Weeknight": false,
			"Weekend": false,
			"Gluten free": false,
			"Bread": false,
			"Dairy free": false,
			"White meat": false,
		},
		cuisine: "Any",
		serves: "Any",
		acknowledgement: ""
	},
	newUserDetails: {
		first_name: "",
		last_name: "",
		username: "",
		e_mail: "",
		password: "",
		password_confirmation: "",
		country: "United States",
		image_url: "",
		profile_text: ""
	},
	loginUserDetails: {
		e_mail: "",
		password: ""
	},
	chefs: {
		all_chefs: [],
		followed: [],
		most_liked_chefs: [],
		most_made_chefs: [],
		chef_followees: [],
		chef_followers: []
	},
	chefs_details: {},
	filter_settings: {
		"Breakfast": false,
		"Lunch": false,
		"Dinner": false,
		"Chicken": false,
		"Red meat": false,
		"Seafood": false,
		"Vegetarian": false,
		"Salad": false,
		"Vegan": false,
		"Soup": false,
		"Dessert": false,
		"Side": false,
		"Whole 30": false,
		"Paleo": false,
		"Freezer meal": false,
		"Keto": false,
		"Weeknight": false,
		"Weekend": false,
		"Gluten free": false,
		"Bread": false,
		"Dairy free": false,
		"White meat": false,
	},
	filterCuisines: {
		all: "Any",
		chef: "Any",
		chef_feed: "Any",
		chef_liked: "Any",
		chef_made: "Any",
		global_ranks: "Any",
		most_liked: "Any",
		most_made: "Any"
	},
	serves: "Any",
	stayLoggedIn: false,
}

export const middleware = compose( //exported ONLY for unit testing
	applyMiddleware(ReduxThunk)
)

export const store = createStore(reducer, initialState, middleware)
