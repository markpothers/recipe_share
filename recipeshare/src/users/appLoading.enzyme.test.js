/**
 * @jest-environment jsdom
 */

// stock imports always required for enzyme testing
import React from 'react';
import { mount } from 'enzyme'
import { act } from 'react-dom/test-utils';
import { createSerializer } from 'enzyme-to-json';
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
import toJson from 'enzyme-to-json';
import { findByTestID } from '../auxTestFunctions/findByTestId'

// suite-specific imports
import { createStore } from 'redux';
import { Provider } from 'react-redux'
import { initialState, middleware } from '../redux/store'
import reducer from '../redux/reducer.js'
import AppLoading from './appLoading.js'
import { Image } from 'react-native'
import AsyncStorage from '@react-native-async-storage/async-storage'
// import { loadToken } from '../auxFunctions/saveLoadToken'

// manual mocks
// jest.mock('../auxFunctions/saveLoadToken')

describe('AppLoading', () => {

	let component
	let instance
	let store
	let mockSetLoadedAndLoggedIn

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	})

	beforeEach(async () => {
		// console.log('runs before every test')
		store = createStore(reducer, initialState, middleware)
		mockSetLoadedAndLoggedIn = jest.fn()

		await act(async () => {
			component = await mount(
				<AppLoading
					setLoadedAndLoggedIn={mockSetLoadedAndLoggedIn}
				/>,
				{
					wrappingComponent: Provider,
					wrappingComponentProps: {
						store: store
					}
				}
			)
		})
		component.setProps({})
		instance = component.children().instance()
	})

	afterEach(async () => {
		// console.log('runs after each test')
		AsyncStorage.setItem.mockClear() //forget calls to this method between tests
		AsyncStorage.getItem.mockClear()
		AsyncStorage.clear() //clear out the contents of AsyncStorage between tests
	})

	afterAll(() => {
		// console.log('runs after all tests have completed')
	})

	describe('rendering', () => {

		test('renders fully; has 1 logo', async () => {
			let json = toJson(component)
			expect(json).toMatchSnapshot()
			let logo = findByTestID(component, Image, 'yellowLogo')
			expect(logo).toBeTruthy()
		})

	})

	describe('mount-comes', () => {

		test('mounts with no chef saved', async () => {
			await AsyncStorage.getItem.mockResolvedValueOnce(null)
			expect(AsyncStorage.getItem).toHaveBeenCalled()
			expect(AsyncStorage.getItem).toHaveBeenCalledWith('chef')
			expect(mockSetLoadedAndLoggedIn).toHaveBeenCalled()
			expect(mockSetLoadedAndLoggedIn).toHaveBeenCalledWith({ loaded: true, loggedIn: false })
		})

		test('mounts with a chef saved', async () => {
			expect(instance.props.stayingLoggedIn).toEqual(false)
			expect(instance.props.loggedInChef).toEqual({
				id: "",
				username: "",
				auth_token: "",
				image_url: "",
				is_admin: false
			})
			let chef = {
				"auth_token": "testAuthToken",
				"country": "United States",
				"created_at": "2020-11-16T22:50:15.986Z",
				"deactivated": false,
				"e_mail": "test@email.com",
				"first_name": null,
				"hex": "",
				"id": 22,
				"image_url": "",
				"is_admin": true,
				"is_member": false,
				"last_name": null,
				"password_is_auto": false,
				"profile_text": "mock profile text",
				"username": "my test username",
			}
			await AsyncStorage.getItem.mockResolvedValueOnce(JSON.stringify(chef))
			await act(async () => {
				component = await mount(
					<AppLoading
						setLoadedAndLoggedIn={mockSetLoadedAndLoggedIn}
					/>,
					{
						wrappingComponent: Provider,
						wrappingComponentProps: {
							store: store
						}
					}
				)
			})
			component.setProps({})
			instance = component.children().instance()
			component.update()
			expect(instance.props.stayingLoggedIn).toEqual(true) //because stay logged in was called
			expect(instance.props.loggedInChef).toEqual({
				id: chef.id,
				e_mail: chef.e_mail,
				username: chef.username,
				auth_token: chef.auth_token,
				image_url: chef.image_url,
				is_admin: chef.is_admin,
			}) // because update logged in chef in state was called
			expect(AsyncStorage.getItem).toHaveBeenCalled()
			expect(AsyncStorage.getItem).toHaveBeenCalledWith('chef')
			expect(mockSetLoadedAndLoggedIn).toHaveBeenCalled()
			expect(mockSetLoadedAndLoggedIn).toHaveBeenCalledWith({ loaded: true, loggedIn: true })
		})

	})

})