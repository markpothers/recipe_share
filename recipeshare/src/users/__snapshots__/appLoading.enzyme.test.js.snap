// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoading rendering renders fully; has 1 logo 1`] = `
<Connect(AppLoading)
  setLoadedAndLoggedIn={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "loaded": true,
            "loggedIn": false,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <AppLoading
    e_mail=""
    loggedInChef={
      Object {
        "auth_token": "",
        "id": "",
        "image_url": "",
        "is_admin": false,
        "username": "",
      }
    }
    password=""
    setLoadedAndLoggedIn={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "loaded": true,
              "loggedIn": false,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    stayLoggedIn={[Function]}
    stayingLoggedIn={false}
    updateLoggedInChefInState={[Function]}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <ImageBackground
          imageStyle={Object {}}
          source={1}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Image
                source={1}
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": undefined,
                    },
                    Object {},
                  ]
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": undefined,
                        "width": undefined,
                      },
                      Object {},
                    ]
                  }
                />
              </Image>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "top": 133.4,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "top": 133.4,
                    }
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      Object {
                        "height": 266.8,
                        "width": "75%",
                      }
                    }
                    testID="yellowLogo"
                  >
                    <Image
                      resizeMode="contain"
                      source={1}
                      style={
                        Object {
                          "height": 266.8,
                          "width": "75%",
                        }
                      }
                      testID="yellowLogo"
                    />
                  </Image>
                </View>
              </View>
            </View>
          </View>
        </ImageBackground>
      </View>
    </View>
  </AppLoading>
</Connect(AppLoading)>
`;
